app/client/models/ToggleEnterpriseModel.ts

```diff
@@ -57,27 +55,35 @@ export class ToggleEnterpriseModel extends Disposable {
     if (this.busy.get()) {
       throw new Error("Please wait for the previous operation to complete.");
     }
+    this.busy.set(true);
     try {
-      this.busy.set(true);
       await this._notifier.slowNotification(func());
-    } finally {
+      await this._reloadWhenReady();
+    } catch (err) {
       this.busy.set(false);
+      throw err;
     }
   }
 
   private async _reloadWhenReady() {
-    // Now wait for the server to come back up, and refresh the page.
-    let maxTries = 10;
+    // Now wait about 30 seconds for the server to come back up, and
+    // refresh the page.
+    let maxTries = 30;
     while(maxTries-- > 0) {
       try {
-        await this._configAPI.getValue('edition');
+        const resp = await fetch(getHomeUrl() + "/status?ready=1");
+        if (!resp.ok) {
+          throw new Error(await resp.text());
+        }
         window.location.reload();
         return;
       } catch (err) {
         console.warn("Server not ready yet, will retry", err);
         await delay(1000);
       }
     }
+    // We're done, last step is to reload the page.
```

Comment:
This line is reached when `maxTries` is exhausted, right? Shouldn't we instead throw at this point instead of unconditionally reloading the page? The caller already catches any errors and re-throws them, in addition to setting `busy` to `false`.